pkg/api/scenes.go

```diff
@@ -117,6 +121,50 @@ func (i SceneResource) WebService() *restful.WebService {
 	return ws
 }
 
+func (i SceneResource) createCustomScene(req *restful.Request, resp *restful.Response) {
+	db, _ := models.GetDB()
+	defer db.Close()
+
+	title := req.QueryParameter("title")
+	if (title == "") {
+		log.Error("Title is missing from request!")
+		return
+	}
+
+	var unusedCustomID string
+
+	var lastScene models.Scene
+	err := db.Where("site = ?", "CustomVR").Order("SceneID").Last(&lastScene)
```

Comment:
A few issues with this:
1. The column is named `scene_id`, doesn't this just fail (i.e. return on line 147)?
2. The order is implicitly ascending and `Last` is defined as taking the last record ordered by primary key. This result in a query like `... order by scene_id asc, id desc limit 1`... which selects the first custom scene (`Custom-0`) rather than the last one.
3. Ordering by scene ID to get the last custom scene will stop working at the 11th custom scene because `Custom-9 > Custom-10`.
4. Getting the last scene ID and incrementing it by 1 is going to be problematic if the option to delete custom scenes is added (or if one deletes the scene manually from the database). Say I create `Custom-0`, edit it to add a title and description and then delete it. Then I create a new custom scene which gets `Custom-0` again as its scene ID. Now say I edit this one but only set a title, no description. Next time edits get re-applied after scraping, the new `Custom-0` scene will get the description from the one that was deleted.

(I haven't tested any of this in XBVR, but I *have* tested it in a small standalone test program.)

Reply:
I did do simple tests that this did work with up to 8 scenes, but you are right that sql will probably do string sorting and it will break (remember to backup the database if doing this).
I think the solution to this is to have an integer only this code reads and writes to, so it always gets incremented, even if all other scenes gets deleted.
Im unsure where to store said integer though, as i see it, there are 2 options.
I make a database table with only 1 entry, where said entry is said iterator.
Or, i make a file in the config folder where i store said integer and write to it.

The advantage of the single file is that it seems wasteful to create a db table for such a simple value.
But the advantage of a db entry, is that its more stable and users are less likely to mess with it.

Reply:
Alright, I tested it within XBVR and it does work, even past 11 custom scenes. However it works kind of by accident. `Order("SceneID")` is invalid so GORM *v1* just silently ignores it, so it ends up taking the last scene ordered by `ID` which works just fine since the ID is auto-incremented. GORM *v2*, which I was initially testing with, errors out because of the invalid column name. This accounts for my points 1 to 3.

As for point 4, another option would be to derive the scene ID from the current date and time. Or creating a random one and checking that it doesn't already exist in either the `scenes` table or the `actions` table.

Reply:
Alright, so i now made the scene id based on the current time, the norm seemed to be to have a number as id, so i made it into a number (currently its time down to 2 decimals of the millisecond, to attempt to prevent collisions).
Its also accepts a custom id from the user, although this isn't present in the UI.

Reply:
A assume that thumbs up means approved 😀